src/openzaak/components/autorisaties/admin_views.py

```diff
@@ -0,0 +1,271 @@
+                    }
+                )
+            initial.append(_initial)
+
+    elif component == ComponentTypes.brc:
+        besluittype_ids = set(BesluitType.objects.values_list("id", flat=True))
+        relevant_ids = set(related_objs.values())
+
+        _initial = {}
+        if spec:
+            _initial[
+                "related_type_selection"
+            ] = RelatedTypeSelectionMethods.all_current_and_future
+        elif besluittype_ids == relevant_ids:
+            _initial["related_type_selection"] = RelatedTypeSelectionMethods.all_current
+        else:
+            _initial.update(
+                {
+                    "related_type_selection": RelatedTypeSelectionMethods.manual_select,
+                    "besluittypen": relevant_ids,
+                }
+            )
+        initial.append(_initial)
+
+    return initial
+
+
+def get_initial(applicatie: Applicatie) -> List[Dict[str, Any]]:
+    """
+    Figure out the initial data for the formset, showing existing config.
+
+    We group applicatie autorisaties bij (component, scopes) and evaluate
+    if this constitutes one of the "special" options. If so, we can provide
+    this information to the form, presenting it much more condensed to the
+    end user.
+    """
+    initial = []
+
+    autorisatie_specs = {
+        spec.component: spec for spec in applicatie.autorisatie_specs.all()
+    }
+
+    grouped = defaultdict(list)
+    autorisaties = applicatie.autorisaties.all()
+    for autorisatie in autorisaties:
+        key = (
+            autorisatie.component,
+            tuple(sorted(autorisatie.scopes)),
+        )
+        grouped[key].append(autorisatie)
+
+    for (component, _scopes), _autorisaties in grouped.items():
+        component_initial = get_initial_for_component(
+            component, _autorisaties, autorisatie_specs.get(component),
+        )
+
+        initial += [
+            {"component": component, "scopes": list(_scopes), **_initial}
+            for _initial in component_initial
+        ]
+
+    return initial
+
+
+class AutorisatiesView(DetailView):
+    model = Applicatie
+    template_name = "admin/autorisaties/applicatie_autorisaties.html"
+    pk_url_kwarg = "object_id"
+    # set these on the .as_viev(...) call
+    admin_site = None
+    model_admin = None
+
+    # perform permission checks
+    def dispatch(self, request, *args, **kwargs):
+        assert self.admin_site
+        assert self.model_admin
+
+        applicatie = self.get_object()
+        if not self.model_admin.has_change_permission(request, applicatie):
+            raise PermissionDenied()
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        self.object = applicatie = self.get_object()
+        formset = self.get_formset()
+
+        if formset.is_valid():
+            formset.save()
+
+            # TODO: send out notification to NRC!
```

Comment:
Issue created: https://github.com/open-zaak/open-zaak/issues/355

Reply:
Removing the TODO as it's implemented